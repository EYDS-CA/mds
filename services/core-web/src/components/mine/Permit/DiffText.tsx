import React from 'react';
import { Typography } from 'antd';
import { diffWords } from 'diff';

interface DiffTextProps {
    oldText: string;
    newText: string;
}

/**
 * DiffText component to show the difference between two texts using diffWords from js-diff
 * @param oldText the old text
 * @param newText the new text
 * @returns Text with differences highlighted
 */
const DiffText: React.FC<DiffTextProps> = ({ oldText, newText }) => {
    const differences = diffWords(oldText, newText);

    return (
        <Typography.Text>
            {differences.map((part, index) => {
                if (part.added) {
                    return <span key={index} className="diff-added">{part.value}</span>;
                }
                if (part.removed) {
                    return <span key={index} className="diff-removed">{part.value}</span>;
                }
                return <span key={index}>{part.value}</span>;
            })}
        </Typography.Text>
    );
};

export default DiffText;