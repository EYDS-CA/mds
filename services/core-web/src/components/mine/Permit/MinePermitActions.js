import React, { Component } from "react";
import { connect } from "react-redux";
import { Dropdown, Menu, Button } from "antd";
import PropTypes from "prop-types";
import { EDIT_OUTLINE_VIOLET } from "@/constants/assets";
import { DownloadOutlined } from "@ant-design/icons";
import CustomPropTypes from "@/customPropTypes";
import { getDocumentDownloadToken } from "@common/utils/actionlessNetworkCalls";
import { setPermitAmendmentDownloadState } from "@common/actionCreators/permitActionCreator";
import { bindActionCreators } from "redux";

const propTypes = {
  permitAmendment: CustomPropTypes.permitAmendment.isRequired,
  permit: CustomPropTypes.permit.isRequired,
  openEditAmendmentModal: PropTypes.func.isRequired,
  setPermitAmendmentDownloadState: PropTypes.func.isRequired,
};

const defaultProps = {};

export class MinePermitActions extends Component {
  state = {
    cancelDownload: false,
  };

  cancelDownload = () => {
    this.setState({ cancelDownload: true });
  };

  downloadDocument = (url) => {
    const a = document.createElement("a");
    a.href = url.url;
    a.download = url.filename;
    a.style.display = "none";
    document.body.append(a);
    a.click();
    a.remove();
  };

  waitFor = (conditionFunction) => {
    const poll = (resolve) => {
      if (conditionFunction()) resolve();
      else setTimeout(() => poll(resolve), 400);
    };

    return new Promise(poll);
  };

  downloadDocumentPackage = () => {
    const docURLS = [];

    const permitAmendmentSubmissions = this.props.permit.permit_amendments[0].related_documents.map(
      (doc) => ({
        key: doc.mine_report_submission_guid,
        documentManagerGuid: doc.document_manager_guid,
        filename: doc.document_name,
      })
    );
    console.log("this.props.permit", this.props.permit);
    console.log("this.props.permitAmendment", this.props.permitAmendment);

    const totalFiles = permitAmendmentSubmissions.length;
    if (totalFiles === 0) {
      return;
    }

    console.log("permitAmendmentSubmissions", permitAmendmentSubmissions);
    console.log("totalFiles", totalFiles);

    permitAmendmentSubmissions.forEach((doc) =>
      getDocumentDownloadToken(doc.documentManagerGuid, doc.filename, docURLS)
    );

    let currentFile = 0;
    this.waitFor(() => docURLS.length === permitAmendmentSubmissions.length).then(async () => {
      // eslint-disable-next-line no-restricted-syntax
      for (const url of docURLS) {
        if (this.state.cancelDownload) {
          this.setState({ cancelDownload: false });
          setPermitAmendmentDownloadState({
            downloading: false,
            currentFile: 0,
            totalFiles: 1,
          });
          notification.success({
            message: "Cancelled file downloads.",
            duration: 10,
          });
          return;
        }
        currentFile += 1;
        setPermitAmendmentDownloadState({
          downloading: true,
          currentFile,
          totalFiles,
        });
        this.downloadDocument(url);
        // eslint-disable-next-line
        await new Promise((resolve) => setTimeout(resolve, 2000));
      }
      notification.success({
        message: `Successfully Downloaded: ${totalFiles} files.`,
        duration: 10,
      });

      setPermitAmendmentDownloadState({
        downloading: false,
        currentFile: 1,
        totalFiles: 1,
      });
    });
  };

  render() {
    console.log("this.props.permitAmendment", this.props.permitAmendment);
    const menu = (
      <Menu>
        <Menu.Item key="0">
          <Button
            ghost
            type="primary"
            className="full"
            onClick={(event) =>
              this.props.openEditAmendmentModal(
                event,
                this.props.permitAmendment,
                this.props.permit
              )
            }
          >
            <div>
              <img src={EDIT_OUTLINE_VIOLET} alt="Edit Report" />
            </div>
          </Button>
        </Menu.Item>
        <Menu.Item key="1">
          <Button
            ghost
            type="primary"
            className="full"
            onClick={() => this.downloadDocumentPackage()}
          >
            <DownloadOutlined className="padding-sm--right icon-sm" />
          </Button>
        </Menu.Item>
      </Menu>
    );

    return (
      <Dropdown overlay={menu} placement="bottomLeft">
        <Button type="secondary">Actions</Button>
      </Dropdown>
    );
  }
}

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      setPermitAmendmentDownloadState,
    },
    dispatch
  );

export default connect(null, mapDispatchToProps)(MinePermitActions);
