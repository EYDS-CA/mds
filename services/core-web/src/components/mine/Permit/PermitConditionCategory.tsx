import FormWrapper from "@mds/common/components/forms/FormWrapper";
import RenderField from "@mds/common/components/forms/RenderField";
import RenderSubmitButton from "@mds/common/components/forms/RenderSubmitButton";
import { FORM } from "@mds/common/constants/forms";
import { IPermitConditionCategory } from "@mds/common/interfaces";
import { Button, Col, Form, Popconfirm, Row, Tooltip, Typography } from "antd";
import React, { useState } from "react";
import { Field } from "redux-form";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faArrowDown, faArrowUp, faCheck, faTrash, faXmark } from "@fortawesome/pro-light-svg-icons";
import PermitConditionCategorySelector from "./PermitConditionCategorySelector";
import { required } from "@mds/common/redux/utils/Validate";
import { reset } from 'redux-form';
import { useDispatch } from "react-redux";

export interface IPermitConditionCategoryProps {
  onChange: (category: IPermitConditionCategory) => void | Promise<void>;
  onDelete: (category: IPermitConditionCategory) => void | Promise<void>;
  moveUp: (category: IPermitConditionCategory) => void | Promise<void>;
  moveDown: (category: IPermitConditionCategory) => void | Promise<void>;
  category: IPermitConditionCategory;
  conditionCount: number;
  currentPosition: number;
  categoryCount: number;
}

export const EditPermitConditionCategoryInline = (props: IPermitConditionCategoryProps) => {
  const [isEditMode, setIsEditMode] = useState(false);

  const dispatch = useDispatch();
  const formName = `${FORM.INLINE_EDIT_PERMIT_CONDITION_CATEGORY}}-${props.category.condition_category_code}`;
  const enableEditMode = (evt) => {
    evt.stopPropagation();
    evt.preventDefault();
    setIsEditMode(true);
  };

  const handleSubmit = (cat) => {
    props.onChange(cat);
    setIsEditMode(false);
  }

  const handleDelete = (cat) => {
    props.onDelete(cat);
  }

  const cancel = (evt) => {
    evt.stopPropagation();


    dispatch(reset(formName));
    setIsEditMode(false);
  }


  if (!isEditMode) {
    return (
      <Tooltip title="Click to edit">
        <div onClick={enableEditMode} onKeyDown={enableEditMode}>
          <Typography.Title style={{ marginBottom: 0 }} level={3}>{props.category.step ? `${props.category.step}. ` : ''}{props.category.description} ({props.conditionCount})</Typography.Title>
        </div>
      </Tooltip>
    );
  }

  return (
    <FormWrapper scrollOnToggleEdit={false} name={formName} onSubmit={handleSubmit} initialValues={props.category} isEditMode={isEditMode}>
      <Row style={{ gap: '0.5em' }}>
        <Col flex-shrink="1" style={{ maxWidth: '40px' }}>
          <Field name="step" component={RenderField} required={true} validate={[required]} style={{ marginRight: 0, }} />
        </Col>
        <Col>
          <PermitConditionCategorySelector showLabel={false} />
        </Col>
        <Col flex="auto" style={{ display: 'flex', gap: '0.5em' }}>
          <Button
            className="icon-button-container"
            style={{ marginRight: 0 }}
            onClick={cancel}
            type="primary"
            icon={<FontAwesomeIcon icon={faXmark} />}
          />

          <RenderSubmitButton buttonText="" buttonProps={{ "aria-label": "Confirm", className: "icon-button-container", style: { marginRight: 0, marginLeft: 0 }, icon: <FontAwesomeIcon icon={faCheck} /> }} />

          <Popconfirm
            disabled={props.conditionCount > 0}
            placement="topRight"
            title={
              <>
                <Typography.Paragraph>Are you sure you want to delete {props.category.description}?</Typography.Paragraph>
                <Typography.Paragraph>This action cannot be undone.</Typography.Paragraph>
              </>
            }
            onConfirm={() => handleDelete(props.category)}
            okText="Yes, Delete Category"
            cancelText="No"
          >
            <Button
              disabled={props.conditionCount > 0}
              danger={true}
              icon={<FontAwesomeIcon icon={faTrash} />}
              aria-label="Delete Category" />
          </Popconfirm>

          <Button
            disabled={props.currentPosition <= 0}
            onClick={(event) => {
              event.stopPropagation();
              props.moveUp(props.category);
            }}
            type="default"
            aria-label="Move Category Up"
            icon={<FontAwesomeIcon icon={faArrowUp} />}
          />
          <Button
            style={{ marginLeft: 0 }}
            disabled={props.currentPosition >= props.categoryCount - 1}
            aria-label="Move Category Down"
            onClick={(event) => {
              event.stopPropagation();
              props.moveDown(props.category);
            }}
            icon={<FontAwesomeIcon icon={faArrowDown} />}
          />
        </Col>
      </Row>
    </FormWrapper >
  )
};
