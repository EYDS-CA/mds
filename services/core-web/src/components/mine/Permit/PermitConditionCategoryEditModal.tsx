import React, { FC } from "react";
import { Col, Form, Row } from "antd";
import { FORM } from "@mds/common/constants/forms";
import FormWrapper from "@mds/common/components/forms/FormWrapper";
import RenderCancelButton from "@mds/common/components/forms/RenderCancelButton";
import RenderSubmitButton from "@mds/common/components/forms/RenderSubmitButton";
import { Field } from "redux-form";
import { maxLength, required } from "@mds/common/redux/utils/Validate";
import RenderField from "@mds/common/components/forms/RenderField";
import { IPermitConditionCategory } from "@mds/common/interfaces";
import PermitConditionCategorySelector from "./PermitConditionCategorySelector";

interface PermitConditionCategoryEditModalProps {
  handleSubmit(category: IPermitConditionCategory): Promise<void>;
}

const PermitConditionCategoryEditModal: FC<PermitConditionCategoryEditModalProps> = ({ handleSubmit }) => {
  return (
    <FormWrapper name={FORM.ADD_PERMIT_CONDITION_CATEGORY} isModal onSubmit={handleSubmit}>
      <Row gutter={6}>
        <Col span={24}>
          <Form.Item>
            <Field
              id="step"
              name="step"
              label="Category Reference Number"
              validate={[required, maxLength(2)]}
              required
              component={RenderField}
            />
          </Form.Item>
        </Col>
        <Col span={24}>
          <PermitConditionCategorySelector />
        </Col>
      </Row>

      <div className="ant-modal-footer">
        <RenderCancelButton />
        <RenderSubmitButton buttonText="Add Category" disableOnClean={false} />
      </div>
    </FormWrapper>
  );
};

export default PermitConditionCategoryEditModal;
