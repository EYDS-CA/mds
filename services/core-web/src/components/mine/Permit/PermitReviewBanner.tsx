import { faInfoCircle, faWarning } from "@fortawesome/pro-regular-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { Col, Row } from "antd";
import React, { FC } from "react";

interface PermitReviewBannerProps {
    isExtracted: boolean;
    height: number;
    isReviewComplete: boolean;
};

export const PermitReviewBanner: FC<PermitReviewBannerProps> = ({
    isExtracted,
    height,
    isReviewComplete
}) => {
    const completeParams = {
        text: "Conditions were extracted using AI and have been reviewed and verified.",
        className: "complete",
        icon: faInfoCircle
    }

    const incompleteParams = {
        text: "Conditions and their report requirements have been extracted using AI and require review and verification.",
        className: "incomplete",
        icon: faWarning
    };

    const draftedCoreParams = {
        ...completeParams,
        text: "This permit was drafted and issued in Core; conditions cannot be modified."
    };

    const aiParams = isReviewComplete ? completeParams : incompleteParams;
    const paramsToUse = isExtracted ? aiParams : draftedCoreParams;

    return (<>
        <Row
            justify="center"
            align="middle"
            className={`permit-status-banner permit-status-banner--${paramsToUse.className}`}
            style={{
                height,
            }}
        >
            <Col>{<FontAwesomeIcon icon={paramsToUse.icon} />}</Col>
            <Col>{paramsToUse.text}</Col>
        </Row>
        <div
            style={{
                height
            }}
        ></div>
    </>);
};