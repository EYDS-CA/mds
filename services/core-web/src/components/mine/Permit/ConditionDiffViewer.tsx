import React, { FC, useEffect } from 'react';
import { Row, Col, Typography } from 'antd';
import { IPermitAmendment, IPermitCondition, IPermitConditionChangeType, IPermitConditionComparison } from '@mds/common/interfaces';
import { formatPermitConditionStep } from '@mds/common/utils/helpers';
import {
    fetchPermitConditionDiff,
    getPermitConditionDiff,
} from '@mds/common/redux/slices/permitConditionDiffSlice';
import { useAppDispatch, useAppSelector } from '@mds/common/redux/rootState';
import DiffText from './DiffText';

interface Props {
    currentCondition: IPermitCondition;
    previousCondition: IPermitCondition;
    mineGuid: string;
    permitGuid: string;
    latestAmendment: IPermitAmendment;
    previousAmendment?: IPermitAmendment;
}

const ConditionDiffViewer: FC<Props> = ({
    currentCondition,
    previousCondition,
    mineGuid,
    permitGuid,
    latestAmendment,
    previousAmendment,
}) => {
    const dispatch = useAppDispatch();

    const diffs: IPermitConditionComparison[] = useAppSelector(state => getPermitConditionDiff(state, latestAmendment?.permit_amendment_guid));

    useEffect(() => {
        dispatch(fetchPermitConditionDiff({ mineGuid, permitGuid, amendmentGuid: latestAmendment?.permit_amendment_guid }));
    }, [mineGuid, permitGuid, latestAmendment?.permit_amendment_guid]);

    const getComparisonForCondition = (conditionGuid: string) => {
        return diffs?.find(diff => diff?.condition_guid === conditionGuid);
    };

    const getComparisonForPreviousCondition = (conditionGuid: string) => {
        return diffs?.find(diff => diff.previous_condition_guid === conditionGuid);
    };

    const findCondition = (permit_condition_guid: string, conditions: IPermitCondition[]) => {
        const findConditionRecursive = (guid: string, condition: IPermitCondition): IPermitCondition | null => {
            if (condition?.permit_condition_guid === guid) {
                return condition;
            }

            if (condition.sub_conditions) {
                for (const sub of condition.sub_conditions) {
                    const found = findConditionRecursive(guid, sub);
                    if (found) return found;
                }
            }
            return null;
        };

        for (const condition of conditions) {
            const found = findConditionRecursive(permit_condition_guid, condition);
            if (found) return found;
        }
        return null;
    }

    const renderCondition = (condition: IPermitCondition, isPrevious: boolean = false, level = 0) => {
        const comparison = isPrevious
            ? getComparisonForPreviousCondition(condition?.permit_condition_guid)
            : getComparisonForCondition(condition?.permit_condition_guid);

        const getChangeClass = () => {
            if (!comparison) {
                return isPrevious ? 'condition-removed' : 'condition-unchanged';
            }

            if (isPrevious) {
                switch (comparison.change_type) {
                    case IPermitConditionChangeType.MODIFIED:
                    case IPermitConditionChangeType.MOVED:
                        return `condition-${comparison.change_type.toLowerCase()}`;
                    default:
                        return 'condition-unchanged';
                }
            }

            return `condition-${comparison.change_type.toLowerCase()}`;
        };

        // Find related condition for diff text
        const relatedCondition = isPrevious
            ? findCondition(comparison?.condition_guid || '', currentCondition.sub_conditions || [])
            : findCondition(comparison?.previous_condition_guid || '', previousAmendment?.conditions || []);

        return (
            <>
                <div className={getChangeClass()}>
                    <div className="condition-text" style={{ marginLeft: `${level * 20}px`, marginBottom: '8px' }}>
                        <Row wrap={false} align="top" className="condition-content">
                            <Col className="step-column" style={{ flexShrink: 0 }}>
                                <Typography.Text className="view-item-value">
                                    {formatPermitConditionStep(condition.step)} &nbsp;
                                </Typography.Text>
                            </Col>
                            <Col className="condition-column">
                                {comparison?.change_type === IPermitConditionChangeType.MODIFIED ? (
                                    <DiffText
                                        oldText={relatedCondition?.condition || ''}
                                        newText={condition.condition}
                                    />
                                ) : (
                                    <Typography.Text className="view-item-value">
                                        {condition.condition}
                                    </Typography.Text>
                                )}
                            </Col>
                        </Row>
                    </div>
                </div>
                {condition.sub_conditions?.map((sub) => renderCondition(sub, isPrevious, level + 1))}
            </>
        );
    };
    return (
        <div className="diff-container">
            <Row gutter={16}>
                <Col span={12}>
                    <Typography.Title level={4}>Previous Version</Typography.Title>
                    {renderCondition(previousCondition, true)}
                </Col>
                <Col span={12}>
                    <Typography.Title level={4}>Current Version</Typography.Title>
                    {renderCondition(currentCondition)}
                </Col>
            </Row>
        </div>
    );
};

export default ConditionDiffViewer;