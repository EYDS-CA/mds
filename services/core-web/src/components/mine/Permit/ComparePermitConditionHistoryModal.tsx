import React, { FC } from "react";
import { Col, Row, Typography } from "antd";
import { IPermitAmendment, IPermitCondition } from "@mds/common/interfaces";
import FormWrapper from "@mds/common/components/forms/FormWrapper";
import ConditionDiffViewer from "./ConditionDiffViewer";

interface ComparePermitConditionHistoryModalProps {
    currentAmendmentCondition: IPermitCondition;
    previousAmendmentCondition: IPermitCondition;
    mineGuid: string;
    permitGuid: string;
    latestAmendment: IPermitAmendment;
}

const ComparePermitConditionHistoryModal: FC<ComparePermitConditionHistoryModalProps> = (props) => {

    return (
        <FormWrapper name={"test"} isModal onSubmit={() => { }}>
            <Row gutter={6}>
                <Col span={24}>
                    <Typography.Title level={3}>Compare Conditions</Typography.Title>
                </Col>
                <Col span={24}>
                    <ConditionDiffViewer
                        mineGuid={props.mineGuid}
                        permitGuid={props.permitGuid}
                        latestAmendment={props.latestAmendment}
                        currentCondition={props.currentAmendmentCondition}
                        previousCondition={props.previousAmendmentCondition}
                    />
                </Col>
            </Row>
            <div className="ant-modal-footer" />
        </FormWrapper>
    );
};

export default ComparePermitConditionHistoryModal;