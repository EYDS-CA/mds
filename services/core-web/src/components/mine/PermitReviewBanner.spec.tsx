import React from "react";
import { render } from "@testing-library/react";
import { PermitReviewBanner } from "./Permit/PermitReviewBanner";

it("renders correctly", () => {
    const height = 30;
    const { container } = render(<>
        <p>AI: review complete</p>
        <PermitReviewBanner height={height} isReviewComplete={true} isExtracted={true} />
        <p>AI: requires review</p>
        <PermitReviewBanner height={height} isReviewComplete={false} isExtracted={true} />
        <p>Drafted in core, cannot be modified</p>
        <PermitReviewBanner height={height} isReviewComplete={true} isExtracted={false} />
    </>);

    expect(container).toMatchSnapshot();
});