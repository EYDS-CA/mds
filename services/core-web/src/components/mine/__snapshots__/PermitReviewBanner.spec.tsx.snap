// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <p>
    AI: review complete
  </p>
  <div
    class="ant-row ant-row-center ant-row-middle permit-status-banner permit-status-banner--complete"
    style="height: 30px;"
  >
    <div
      class="ant-col"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-circle-info "
        data-icon="circle-info"
        data-prefix="far"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 48a208 208 0 1 1 0 416 208 208 0 1 1 0-416zm0 464A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM216 336c-13.3 0-24 10.7-24 24s10.7 24 24 24h80c13.3 0 24-10.7 24-24s-10.7-24-24-24h-8V248c0-13.3-10.7-24-24-24H216c-13.3 0-24 10.7-24 24s10.7 24 24 24h24v64H216zm40-144a32 32 0 1 0 0-64 32 32 0 1 0 0 64z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="ant-col"
    >
      Conditions were extracted using AI and have been reviewed and verified.
    </div>
  </div>
  <div
    style="height: 30px;"
  />
  <p>
    AI: requires review
  </p>
  <div
    class="ant-row ant-row-center ant-row-middle permit-status-banner permit-status-banner--incomplete"
    style="height: 30px;"
  >
    <div
      class="ant-col"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-triangle-exclamation "
        data-icon="triangle-exclamation"
        data-prefix="far"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M248.4 84.3c1.6-2.7 4.5-4.3 7.6-4.3s6 1.6 7.6 4.3L461.9 410c1.4 2.3 2.1 4.9 2.1 7.5c0 8-6.5 14.5-14.5 14.5H62.5c-8 0-14.5-6.5-14.5-14.5c0-2.7 .7-5.3 2.1-7.5L248.4 84.3zm-41-25L9.1 385c-6 9.8-9.1 21-9.1 32.5C0 452 28 480 62.5 480h387c34.5 0 62.5-28 62.5-62.5c0-11.5-3.2-22.7-9.1-32.5L304.6 59.3C294.3 42.4 275.9 32 256 32s-38.3 10.4-48.6 27.3zM288 368a32 32 0 1 0 -64 0 32 32 0 1 0 64 0zm-8-184c0-13.3-10.7-24-24-24s-24 10.7-24 24v96c0 13.3 10.7 24 24 24s24-10.7 24-24V184z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="ant-col"
    >
      Conditions and their report requirements have been extracted using AI and require review and verification.
    </div>
  </div>
  <div
    style="height: 30px;"
  />
  <p>
    Drafted in core, cannot be modified
  </p>
  <div
    class="ant-row ant-row-center ant-row-middle permit-status-banner permit-status-banner--complete"
    style="height: 30px;"
  >
    <div
      class="ant-col"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-circle-info "
        data-icon="circle-info"
        data-prefix="far"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 48a208 208 0 1 1 0 416 208 208 0 1 1 0-416zm0 464A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM216 336c-13.3 0-24 10.7-24 24s10.7 24 24 24h80c13.3 0 24-10.7 24-24s-10.7-24-24-24h-8V248c0-13.3-10.7-24-24-24H216c-13.3 0-24 10.7-24 24s10.7 24 24 24h24v64H216zm40-144a32 32 0 1 0 0-64 32 32 0 1 0 0 64z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="ant-col"
    >
      This permit was drafted and issued in Core; conditions cannot be modified.
    </div>
  </div>
  <div
    style="height: 30px;"
  />
</div>
`;
